compiler/semtypes.nim

```diff
@@ -1474,7 +1474,9 @@ proc semTypeNode(c: PContext, n: PNode, prev: PType): PType =
 
   if c.config.cmd == cmdIdeTools: suggestExpr(c, n)
   case n.kind
-  of nkEmpty: discard
+  of nkEmpty:
+    if n.typ != nil: result = n.typ
```

Comment:
You can simplify this further since `result` is already initialized to `nil` before and if `n.typ` is `nil` nothing changes.

Reply:
Heh, good point. I should have noticed that. Done.